hw/ip/pwm/doc/_index.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "PWM HWIP Technical Specification"
+---
+
+# Overview
+
+This document specifies PWM hardware IP (HWIP) functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification/index.md" >}})
+See that document for integration overview within the broader top-level system.
+
+## Features
+
+- Provides pulse-width modulated (PWM) drive with adjustable duty cycle.
+- Suitable for general-purpose use, but primarily designed for control of tri-color LEDs
+- Provides a parametrizable number of output channels.
+- All output channels are driven at pulse frequency, which is selected by a 16-bit clock divider.
+   - Clock-divide ratio ranges from 2<sup>16</sup> to 2<sup>32</sup>
+- 16-bit resolution in controlling the output duty cycle.
+   - Duty cycle ranges from  0.0015% (on for only one cycle in every 2<sup>16</sup>) to 100% (always on).
+- 16-bit resolution in adjusting the output phase.
+- Independent control of duty-cycle and phase for all channels
+- Provides an optional hardware-controlled blink feature, with a unique enable-bit for each channel.
+- Channel blink rates are all independently controlled using 16-bit frequency dividers.
+- Blink phase and duty-cycle are independently controlled between channels, with 8-bit resolution.
+- Channels can be configured to blink synchronously or independently.
+
+## Description
+
+The PWM IP is primarily designed to drive a parametrizable number of pulse-width modulated outputs.
+All outputs are driven at a common clock frequency, which can be controlled relative to the input bus clock by a 16-bit clock divider.
```

Comment:
@martin-lueker just a note here one of our applications is that pwm must run in low power state.
So it would probably be good to de-couple `bus clock` from `core clock` here.

Reply:
another note is that the `core clock` may be in the KHz range, so I wonder if that changes your duty cycle length, which seems hard coded to 2^16?

That would mean for a 200KHz clock the duty cycle could be 0.3s, not sure if that's too long.

Reply:
Ah, right, I remember you mentioning this.   Can you give an example of another low-power IP?

Reply:
Oh, interesting.   At such low frequency then we may need to go to programmable phase resolution. (Like the blink rate).

One note to help also: Here the duty cycle is not the time between pulses it is the fraction of on-time.  This definition comes from the Fluke webpage:
"Duty cycle is the ratio of time a load or circuit is ON compared to the time the load or circuit is OFF. Duty cycle, sometimes called "duty factor," is expressed as a percentage of ON time. A 60% duty cycle is a signal that is ON 60% of the time and OFF the other 40%."

Reply:
ah okay sounds good.

Reply:
Need to also consider the switching between clocks if both sys clock and slow clock can be the source of the PWM. This is something that need to be done when the module is disabled

Reply:
Hi Nimrod: 
Do we want to be able to switch between clocks?  Do we need that?  
At least for the Xilinx implementation that would consume a third BUFG.  

If it is really needed, we can do that for sure.  I just want to check that it is needed.  Otherwise I can just have two inputs for the two clocks "bus" and "core", and always have the PWM running from the core clock.

Reply:
2 relevant changes: 
- Added a second clock and reset for core vs. system clock
- Made phase variable resolution.

Reply:
Well, it depends on the desired blink frequency range needed. If all can be achieved using one clock source, there is no need. I know some PWM products that have different clock sources - for example, on in the range of MHz and additional one in the KHz. 
As I am not sure what will be the use case here, I am not sure if more than one clock source is needed